smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CodegenVisitor.java

```diff
@@ -139,7 +144,26 @@ void execute() {
         // Generate models that are connected to the service being generated.
         LOGGER.fine("Walking shapes from " + service.getId() + " to find shapes to generate");
         Set<Shape> serviceShapes = new TreeSet<>(new Walker(nonTraits).walkShapes(service));
-        serviceShapes.forEach(shape -> shape.accept(this));
+        Map<String, Shape> shapeMap = new TreeMap<>();
+
+        // Check for colliding shapes and prune non-unique shapes
+        for (Shape shape : serviceShapes) {
+            String shapeReference = shape.getType().toString() + shape.getId().asRelativeReference();
+
+            if (shapeMap.containsKey(shapeReference)) {
+                Shape knownShape = shapeMap.get(shapeReference);
+                if (!shapesMatch(shape, knownShape)) {
+                    throw new CodegenException(("Shape Collision: cannot condense " + shape.toString() + " and "
```

Comment:
No need to use `toString` on the colliding shapes

Reply:
Fixed in cf9d401